user.go

```diff
@@ -1058,6 +1065,12 @@ func (user *User) e2eeEventHandler(rawEvt any) {
 	}
 }
 
+func (user *User) resetWADeviceAndReconnect(reason events.ConnectFailureReason) {
+	user.log.Debug().Int("ConnectFailureReason", int(reason)).Msg("Resetting WADevice and reconnecting")
+	user.WADevice = nil
```

Comment:
I don't think this is enough for 415s, `WADeviceID` also needs to be cleared (otherwise `connectE2EE` will just reload the existing device from the store) and it would probably be a good idea to delete the device completely from the db